### ISSUE TITLE
Can not import books even though the name of the library is specified

### ISSUE THREAD

**Issue by Plexprofile** [duydl/logseq-calibre-metadata#1]
![image](https://github.com/duydl/logseq-calibre-metadata/assets/139643082/4e8a8c0c-873c-4b80-a6d1-921199f56634)
![image](https://github.com/duydl/logseq-calibre-metadata/assets/139643082/9f16c827-17d0-4aac-b991-694dc8e91a59)
![image](https://github.com/duydl/logseq-calibre-metadata/assets/139643082/c7798c23-0912-4a9a-977d-98983d7ec678)
![image](https://github.com/duydl/logseq-calibre-metadata/assets/139643082/fc175217-229d-44fe-be19-3f0aad4462bb)

**Comment by duydl** [comment]
Have you been able to load the book?

**Comment by Plexprofile** [comment]
I can if I add them manually using the macro renderer, but the books wont pop up in the drop list

**Comment by duydl** [comment]
The books used to show in the drop list right, based on your comments on the other issues? 
thb, I could not think of anything as the problem. Perhaps check the settings and title of your book one more time or wait a little longer before you type the next letter. i.e Could you try just type the letter t and wait for about 10 seconds?

**Comment by Plexprofile** [comment]
> The books used to show in the drop list right, based on your comments on the other issues? 
> thb, I could not think of anything as the problem. Perhaps check the settings and title of your book one more time or wait a little longer before you type the next letter. i.e Could you try just type the letter t and wait for about 10 seconds?

That's exactly what I tried for many times, sadly with no success, even after the reinstallation of the plugin

**Comment by duydl** [comment]
![democalibre2](https://github.com/duydl/logseq-calibre-metadata/assets/56506156/f292bf04-1ab6-4c57-a315-d59fb6e8898d)

Press Ctrl+Shift+i to open the developer tool. After an input in the calibre search bar, the console tab will print the link to fetch metadata and the items received. Above is my output. Could you do it and show me yours?

**Comment by Plexprofile** [comment]
Sorry for the late reply, is this what you are looking for?

http://localhost:8080/ajax/books/Main
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
lsp://logseq.io/logs….html?__v__=1.0.0:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
index.html:1 Access to fetch at 'http://localhost:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
localhost:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
﻿

**Comment by duydl** [comment]
That is strange. If there is a CORS problem you probably would not be able to sync the annotations. Could you?
I will ask around the Logseq and Calibre community if anyone else has the problem.

**Comment by Plexprofile** [comment]
Yes I cannot, the error message is displayed as :

http://localhost:8080//book-get-annotations/library/6-epub
index.6e81d3ad.js:108 
 GET http://localhost:8080//book-get-annotations/library/6-epub 404 (Not Found)
VM283:1 Uncaught (in promise) SyntaxError: Unexpected token 'L', "Library 'l"... is not valid JSON
2
index.7c12aa59.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'twelveHourFormat')
    at t (index.7c12aa59.js:1:75871)
    at index.7c12aa59.js:1:76979

**Comment by duydl** [comment]
Hi. Instead of using the localhost could you try using the IP address for home network instead to see if the problem persists: 
The address is shown here
![image](https://github.com/duydl/logseq-calibre-metadata/assets/56506156/ad103da9-4e78-4e2d-9132-8731033fd582)

which is http://192.168.11.2:8081

**Comment by Plexprofile** [comment]
I tried, it didn't work, the error message went:
Access to fetch at 'http://192.168.43.87:8080/ajax/books/Main' from origin 'lsp://logseq.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
192.168.43.87:8080/ajax/books/Main:1 
 Failed to load resource: net::ERR_FAILED
![image](https://github.com/duydl/logseq-calibre-metadata/assets/139643082/79d7ec23-d735-490e-b6de-32601034ac39)

**Comment by duydl** [comment]
Sorry for the trouble. If you found time to try this again, with the new version, try enabling authen login with username and password in Content Server and fill in the corresponding setting in Logseq.

If the problem persists, please create new issue https://github.com/duydl/logseq-calibre-annotation/issues since I migrated all functionalities of this plugin there.
